vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -0,0 +1,188 @@
+public class DateTimePicker extends AbstractField<DateTimePicker, LocalDateTime>
+        implements HasStyle, Focusable<DateTimePicker>, HasSize {
+
+    private final DatePicker datePicker = new DatePicker();
+    private final TimePicker timePicker = new TimePicker();
+    private Locale locale;
+
+    /**
+     * Default constructor.
+     */
+    public DateTimePicker() {
+        this((LocalDateTime) null);
+    }
+
+    /**
+     * Convenience constructor to create a date time picker with a pre-selected date
+     * in current UI locale format.
+     *
+     * @param initialDateTime
+     *            the pre-selected date time in the picker
+     */
+    public DateTimePicker(LocalDateTime initialDateTime) {
+        super(null);
+        if (initialDateTime != null) {
+            setPresentationValue(initialDateTime);
+            updateValue();
+        }
+
+        getElement().addEventListener("value-changed", e -> this.updateValue());
+
+        addToSlot(datePicker, "date-picker");
+        addToSlot(timePicker, "time-picker");
+
+        setLocale(UI.getCurrent().getLocale());
+    }
+
+    /**
+     * Convenience constructor to create a date time picker with a
+     * {@link ValueChangeListener}.
+     *
+     * @param listener
+     *            the listener to receive value change events
+     * @see #addValueChangeListener(HasValue.ValueChangeListener)
+     */
+    public DateTimePicker(
+            ValueChangeListener<ComponentValueChangeEvent<DateTimePicker, LocalDateTime>> listener) {
+        this();
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Convenience constructor to create a date time picker with a pre-selected date
+     * in current UI locale format and a {@link ValueChangeListener}.
+     *
+     * @param initialDateTime
+     *            the pre-selected date time in the picker
+     * @param listener
+     *            the listener to receive value change events
+     * @see #setValue(LocalDateTime)
+     * @see #addValueChangeListener(HasValue.ValueChangeListener)
+     */
+    public DateTimePicker(LocalDateTime initialDateTime,
+            ValueChangeListener<ComponentValueChangeEvent<DateTimePicker, LocalDateTime>> listener) {
+        this(initialDateTime);
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Convenience Constructor to create a date time picker with pre-selected date
+     * and locale setup.
+     *
+     * @param initialDateTime
+     *            the pre-selected date time in the picker
+     * @param locale
+     *            the locale for the date time picker
+     */
+    public DateTimePicker(LocalDateTime initialDateTime, Locale locale) {
+        this(initialDateTime);
+        setLocale(locale);
+    }
+
+    /**
+     * This method should return the value of the field, based on value of the internal fields.
+     *
+     * @return new value of the field.
+     */
+    protected LocalDateTime generateModelValue() {
```

Comment:
This looks like it should be private. Or alternatively removed and logic moved into `updateValue`.

Reply:
Done (logic moved into `updateValue` and extra method removed).